src/blockchain/rpc.rs

```diff
@@ -0,0 +1,518 @@
+            let wallet_txs_chunk = self
+                .0
+                .as_mut()
+                .ok_or(Error::OfflineClient)?
+                .client
+                .list_transactions(None, Some(ltx_count), Some(ltx_skip), Some(true))?;
+
+            for wallet_tx in &wallet_txs_chunk {
+                // Record whether we observed "settled" transaction and should stop iterating
+                if (wallet_tx.info.blockheight.unwrap_or(std::i32::MAX) as u32) < buried_height {
+                    settled_tx_observed = true;
+                }
+
+                // Collect transactions we don't have yet
+                if wallet_txs
+                    .iter()
+                    .all(|wtx| wtx.info.txid != wallet_tx.info.txid)
+                {
+                    // FIXME
+                    wallet_txs.push(wallet_tx.clone())
+                }
+            }
+
+            // Stop fetching if chunk wasn't full or one transaction was "settled"
+            let no_more = wallet_txs_chunk.len() < ltx_count;
+            if no_more || settled_tx_observed {
+                break;
+            }
+
+            // Update RPC cursor for next iteration
+            ltx_skip += ltx_count;
+        }
+
+        // Sort wallet transactions from oldest to newest
+        wallet_txs.sort_by(|a, b| b.info.confirmations.cmp(&a.info.confirmations));
+
+        // Map wallet transactions to full transactions (TODO: batching)
+        let mut txs: Vec<Transaction> = vec![];
+        for wallet_tx in &wallet_txs {
+            let tx = self
+                .0
+                .as_mut()
+                .ok_or(Error::OfflineClient)?
+                .client
+                .get_transaction(&wallet_tx.info.txid, Some(true))?
+                .transaction()?;
+            txs.push(tx);
+        }
+
+        // Map wallet transactions to block heights (TODO: batching)
+        let mut heights: Vec<Option<u32>> = vec![];
+        for wallet_tx in &wallet_txs {
+            let height = match wallet_tx.info.blockhash {
+                Some(blockhash) => Some(
+                    self.0
+                        .as_mut()
+                        .ok_or(Error::OfflineClient)?
+                        .client
+                        .get_block_info(&blockhash)?
+                        .height as u32,
+                ),
+                None => None,
+            };
+            heights.push(height);
+        }
+
+        Ok(txs
+            .iter()
+            .zip(heights.iter())
+            .map(|(tx, height)| (tx.to_owned(), height.to_owned())) // FIXME
+            .collect())
+    }
+}
+
+impl Blockchain for RpcBlockchain {
+    fn offline() -> Self {
+        RpcBlockchain(None)
+    }
+
+    fn is_online(&self) -> bool {
+        self.0.is_some()
+    }
+}
+
+impl OnlineBlockchain for RpcBlockchain {
+    fn get_capabilities(&self) -> HashSet<Capability> {
+        vec![].into_iter().collect()
```

Comment:
I think the `FullHistory` capability should be advertised. It basically means that you can get all the historical transactions of the wallet and not just the ones that have outputs in the UTXO set.

Eventually I think it will make sense to add more options (and sometimes just adapt to the node's configuration) and dinamically add/remove capabilities, like:

- If the node has txindex enabled, then it also supports the `GetAnyTx` capability
- If somebody wants to speed up the sync of his wallet or if the node is pruned, instead of doing a full rescan it could just rescan the utxo set. In that case `FullHistory` should be removed.

Anyways, I think it's fine like this for now, all these improvements could be part of a follow-up pull request.